.github/workflows/pull-gitleaks.yaml

```
@@ -0,0 +1,23 @@
+name: pull-gitleaks
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened, ready_for_review]
```

I think we don't need to run the action when title or body of the PR changed.
```suggestion
    types: [opened, synchronize, reopened, ready_for_review]
```